src/test/fuzz/orphanage.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2020-2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <random.h>
+#include <primitives/transaction.h>
+#include <test/util/script.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <txorphanage.h>
+
+
+FUZZ_TARGET(orphanage)
+{
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+
+    TxOrphanage orphanage;
+    std::set<uint256> orphan_work_set;
+
+    LIMITED_WHILE(fuzzed_data_provider.ConsumeBool(), COINBASE_MATURITY)
+    {
+        // construct transaction
+        const CTransaction tx = [&] {
+            CMutableTransaction tx_mut;
+            tx_mut.nVersion = CTransaction::CURRENT_VERSION;
+            tx_mut.nLockTime = 0;
+            const auto num_in = GetRandInternal(COINBASE_MATURITY);
+            const auto num_out = GetRandInternal(COINBASE_MATURITY);
+            for (unsigned i = 0; i < num_in; i++) {
+                CTxIn in(GetRandHash(), i);
```

Comment:
It might be better to pick either:

* an existing prevout of a previously created transaction
* a random hash

Otherwise you are only ever testing one generation of txs.

Reply:
I think a set of outpoints is better. In your current approach you create `first_tx` -> (second tx) -> (third tx) ... This is a plain chain. Ideally there are some more complicated graphs.